orx/grantlee/0.2/OrxJs.qs

```diff
@@ -0,0 +1,272 @@
+var buffer = '';
+
+function append() {
+    for (var i = 0; i < arguments.length; i++) {
+        if (arguments[i] !== undefined) {
+            buffer += arguments[i] + '\n';
+        }
+    }
+    if (arguments.length == 0) {
+        buffer += '\n';
+    }
+}
+
+function friendlyName(name) {
+    name = name.replace(/[^a-z0-9]/gi, ' ');        // Keep alpha-numeric
+    name = name[0].toUpperCase() + name.slice(1); // Proper case
+    return name;
+}
+
+function getId(name) {
+    // 'big enemy/hard-boss/boss_01' -> ['big', 'enemy', 'hard', 'boss', 'boss', '01']
+    var parts = name.split(/\/|_|-| /g);
+    
+    for (var i = 0; i < parts.length; i++) {
+        parts[i] = friendlyName(parts[i]);
+    }
+    
+    // Optimize any sequences of the same identifier.
+    // E.g. ['Boss', 'Boss', 'Boss'] -> 'Boss'
+    var clean = parts[0];
+    for (var i = 1; i < parts.length; i++) {
+        if (parts[i] !== parts[i - 1]) {
+            clean += parts[i];
+        }
+    }
+    
+    // Now we have an identifier suitable for .ini files.
+    // E.g. 'BigEnemyHardBoss01'
```

Comment:
Is there a reason we're not simply using the input name instead?

Reply:
Input name is a path that tends is less elegant when presented in an `ini` file. To me, it was confusing. Using the same conventions as seen in the tutorials, it was more readable. But yes, this will be controlled by a new setting.